analyzer/analyzer.go

```diff
@@ -135,21 +137,34 @@ func NewAnalyzer(disabledAnalyzers []Type) Analyzer {
 	return Analyzer{
 		drivers:       drivers,
 		configDrivers: configDrivers,
+		disabled:      disabledAnalyzers,
 	}
 }
 
-func (a Analyzer) AnalyzerVersions() map[string]int {
-	versions := map[string]int{}
-	for _, d := range a.drivers {
-		versions[string(d.Type())] = d.Version()
+// AnalyzerVersions returns analyzer version identifier used for cache suffixes.
+// e.g. alpine: 1, amazon: 3, debian: 2 => 132
+// When the amazon analyzer is disabled => 102
+func (a Analyzer) AnalyzerVersions() string {
+	var versions string
+	for _, driver := range analyzers {
+		if isDisabled(driver.Type(), a.disabled) {
+			versions += "0"
+			continue
+		}
+		versions += fmt.Sprint(driver.Version())
```

Comment:
The implementation of AnalyzerVersions depends on the import order.
This is uniquely determined by `go fmt` , but may be affected by changes in package names, etc.

There are several ways to do solution, I can't offer a good this... 
I think we should leave a comment here and Trivy for now.